File: 2021/10/03/main.go
package main

func canJump(nums []int) bool {
	r := len(nums) - 1
	for i := len(nums) - 1; i >= 0; i-- {
		if i+nums[i] >= r {
			r = i
		}
	}

	return r == 0
}

File: 2021/10/03/main.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        r = len(nums) - 1
        
        for i in range(len(nums)-1, -1, -1):
            if i + nums[i] >= r:
                r = i
        
        return r == 0